packages/frontend/web/src/components/form-cocktail/ToppingPart.tsx

```diff
@@ -1,17 +1,106 @@
-import type { ToppingPartProps } from '@app/types';
+import { useEffect, useState } from 'react';
 
-const toppings = ['shrimps', 'lemon', 'mint'];
+import type { Topping, ToppingPartProps } from '@app/types';
 
 export default function ToppingPart({
   register,
   selectedTopping,
+  selectedIngredient,
+  selectedAlcohol,
   handleToppingChange,
   errors,
 }: ToppingPartProps) {
+  const [toppings, setToppings] = useState<Topping[]>([]);
+
+  const allFlavours = [selectedAlcohol?.flavour, selectedIngredient?.flavour];
+  const flavoursCount = {
+    fruity: 0,
+    spicy: 0,
+    herbaceous: 0,
+    floral: 0,
+    woody: 0,
+    bitter: 0,
+    sweet: 0,
+    salty: 0,
+    sour: 0,
+    neutral: 0,
+  };
+
+  for (const flavour of allFlavours) {
+    switch (flavour) {
+      case 'fruity': {
+        flavoursCount.fruity += 1;
+        break;
+      }
+      case 'spicy': {
+        flavoursCount.spicy += 1;
+        break;
+      }
+      case 'herbaceous': {
+        flavoursCount.herbaceous += 1;
+        break;
+      }
+      case 'floral': {
+        flavoursCount.floral += 1;
+        break;
+      }
+      case 'woody': {
+        flavoursCount.woody += 1;
+        break;
+      }
+      case 'bitter': {
+        flavoursCount.bitter += 1;
+        break;
+      }
+      case 'sweet': {
+        flavoursCount.sweet += 1;
+        break;
+      }
+      case 'salty': {
+        flavoursCount.salty += 1;
+        break;
+      }
+      case 'sour': {
+        flavoursCount.sour += 1;
+        break;
+      }
+      case 'neutral': {
+        flavoursCount.neutral += 1;
+        break;
+      }
+    }
+  }
```

Comment:
Tu peux transformer tout ça en 1 reduce
Avec un genre de `flavoursCount = [selectedAlcohol?.flavour, selectedIngredient?.flavour].reduce()`

Reply:
C'était l'idée mais : `Array#reduce()` is not allowedeslint[unicorn/no-array-reduce](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v49.0.0/docs/rules/no-array-reduce.md)

Reply:
Tu crois aux licornes toi ?
Tu peux ajouter `"unicorn/no-array-reduce": "off",` dans le fichier `.eslintrc.cjs` (à placer dans l'ordre alphabétique) avec les autres règles

Reply:
Je crois aux yummyCorns. Ah bon je ne pensais pas qu'on pouvais péter tout ton paramétrage, mais avec plaisir !

Reply:
Je vous autorise à rajouter cette ligne 👍